app/code/core/Mage/Adminhtml/controllers/Sales/Order/CreditmemoController.php

```diff
@@ -295,6 +295,16 @@ public function saveAction()
                     }
                 }
 
+                $succesubscribemsg = '';
+                if (isset($data['newsletter_customer_unsubscribe'])) {
+                    $order = $creditmemo->getOrder();
+                    $email = (string)$order->getBillingAddress()->getEmail();
+                    if (!$email) $email = $order->getCustomerEmail();
+                    if (Mage::getModel('newsletter/subscriber')->loadByEmail($email)->unsubscribe()) {
```

Comment:
I think it would Be good to check if the subscriber was actually loaded before calling `unsubscribe` to avoid an exception thrown.